### PR TITLE
testing/libpuzzle: new aport

### PR DESCRIPTION
**Pull request by boobooking**
The Puzzle library is designed to find visually similar images

CLI utility and PHP7 module in separate packages.

### CODE REVIEW

File: testing/php7-libpuzzle/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# Contributor: Andreas Sundqvist <andreas at forme.se>
+# Maintainer: Timofey Bychkov <boobooin@gmail.com>
+pkgname=php7-libpuzzle
+pkgver=0.11
+pkgrel=0
+pkgdesc="PHP module of the Puzzle library. Library is designed to find visually similar images."
+url="http://libpuzzle.pureftpd.org/project/libpuzzle"
+arch="all"
+license="BSD"
+depends="libpuzzle libgd"
```
Comment (andypost) [inline testing/php7-libpuzzle/APKBUILD]: looks strange when no `libgd-dev` used in build

Comment (boobooking) [reply testing/php7-libpuzzle/APKBUILD]: Actually we don't need this dependency for php module.
There is libpuzzle as dependency, which was build with gd-dev

File: testing/libpuzzle/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+subpackages="$pkgname-doc"
+pkgver=0.11
+pkgrel=0
+pkgdesc="The Puzzle library is designed to find visually similar images."
+url="http://libpuzzle.pureftpd.org/project/libpuzzle"
+arch="all"
+license="BSD"
+depends="libgd"
+makedepends="gd-dev"
+source="https://download.pureftpd.org/pub/pure-ftpd/misc/libpuzzle/releases/libpuzzle-0.11.tar.gz"
```
[the remark this answers is not in the document]
Comment (boobooking) [reply testing/libpuzzle/APKBUILD]: I have fixed that for both packages: libpuzzle and php7-libpuzzle
Thanks!

File: testing/libpuzzle/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,53 @@
+# Developer: Frank DENIS <j at pureftpd.org>
+# Maintainer: Timofey Bychkov <boobooin@gmail.com>
+pkgname=libpuzzle
+subpackages="$pkgname-doc"
+pkgver=0.11
+pkgrel=0
+pkgdesc="The Puzzle library is designed to find visually similar images."
+url="http://libpuzzle.pureftpd.org/project/libpuzzle"
+arch="all"
+license="BSD"
```
Comment (andypost) [inline testing/libpuzzle/APKBUILD]: The puzzle library is free. It is released under the 2-clauses BSD public license

So according spdx it needs change

File: testing/php7-libpuzzle/APKBUILD [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+pkgname=php7-libpuzzle
+pkgver=0.11
+pkgrel=0
+pkgdesc="PHP module of the Puzzle library. Library is designed to find visually similar images."
+url="http://libpuzzle.pureftpd.org/project/libpuzzle"
+arch="all"
+license="BSD"
+depends="libpuzzle"
+makedepends="libpuzzle php7-dev"
+source="https://download.pureftpd.org/pub/pure-ftpd/misc/libpuzzle/releases/libpuzzle-$pkgver.tar.gz 10_php7_support.patch"
```
Comment (andypost) [inline testing/php7-libpuzzle/APKBUILD]: Please split 2 sources in 2 lines

File: testing/php7-libpuzzle/APKBUILD [hunk 5]
```diff
@@ -0,0 +1,36 @@
+# Developer: Frank DENIS <j at pureftpd.org>
+# Contributor: Andreas Sundqvist <andreas at forme.se>
+# Maintainer: Timofey Bychkov <boobooin@gmail.com>
+pkgname=php7-libpuzzle
+pkgver=0.11
+pkgrel=0
+pkgdesc="PHP module of the Puzzle library. Library is designed to find visually similar images."
+url="http://libpuzzle.pureftpd.org/project/libpuzzle"
+arch="all"
+license="BSD"
```
Comment (andypost) [inline testing/php7-libpuzzle/APKBUILD]: Guess it is clause-2

**Comment by boobooking** [comment]
I've commited everything that was requested.
If anything else is needed, please just let me know.

**Comment by boobooking** [comment]
Hi @andypost !

I've commited everything that was requested.
If anything else is needed, please just let me know.

Thanks!

**Comment by boobooking** [comment]
Hi @andypost

I've commited everything that was requested.
If anything else is needed, please just let me know.

Thanks!

**Comment by andypost** [comment]
It still needs squash commits, php module should use check to load extension (see php7-pecl-swoole) also dependencies could use -dev do builder will provide real ones

File: testing/libpuzzle/APKBUILD [hunk 6]
```diff
@@ -0,0 +1,53 @@
+# Developer: Frank DENIS <j at pureftpd.org>
+# Maintainer: Timofey Bychkov <boobooin@gmail.com>
+pkgname=libpuzzle
+subpackages="$pkgname-doc"
+pkgver=0.11
+pkgrel=0
+pkgdesc="The Puzzle library is designed to find visually similar images."
+url="http://libpuzzle.pureftpd.org/project/libpuzzle"
```
Comment (andypost) [inline testing/libpuzzle/APKBUILD]: The repo and site were archived https://github.com/jedisct1/libpuzzle

File: testing/libpuzzle/APKBUILD [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+arch="all"
+license="BSD-2-Clause"
+depends="libgd"
+makedepends="gd-dev"
```
Comment (andypost) [inline testing/libpuzzle/APKBUILD]: that's should be enough without previous line

File: testing/libpuzzle/APKBUILD [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+	install -d "$subpkgdir"
+	install -d "$subpkgdir"/usr/share/man/man3/
+	install -d "$subpkgdir"/usr/share/man/man8/
+	install -d "$subpkgdir"/usr/share/doc/"$pkgname"-"$pkgver"/
+
+	cd "$builddir"
+	install -Dm644 README README-PHP php/libpuzzle/CREDITS php/libpuzzle/LICENSE "$subpkgdir"/usr/share/doc/"$pkgname"-"$pkgver"/
+
+	cd "$builddir"/man
+	gzip puzzle_set.3 libpuzzle.3 puzzle-diff.8
```
Comment (andypost) [inline testing/libpuzzle/APKBUILD]: docs gzipped by packager I bet

File: testing/php7-libpuzzle/APKBUILD [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+source="https://download.pureftpd.org/pub/pure-ftpd/misc/libpuzzle/releases/libpuzzle-$pkgver.tar.gz
+		10_php7_support.patch
+		"
+builddir="$srcdir/libpuzzle-$pkgver"
+options="!check" # Libpuzzle was developed on OpenBSD, and one of the tests doesn't pass because floating point numbers are rounded differently on Linux. It has no implications on usage.
+
+build() {
+	cd "$builddir"/php/libpuzzle
+	phpize
+	./configure
```
Comment (andypost) [inline testing/php7-libpuzzle/APKBUILD]:
```
	phpize7
	./configure --prefix=/usr --with-php-config=php-config7
```

File: testing/php7-libpuzzle/APKBUILD [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+# Contributor: Andreas Sundqvist <andreas at forme.se>
+# Maintainer: Timofey Bychkov <boobooin@gmail.com>
+pkgname=php7-libpuzzle
+pkgver=0.11
+pkgrel=0
+pkgdesc="PHP module of the Puzzle library. Library is designed to find visually similar images."
+url="http://libpuzzle.pureftpd.org/project/libpuzzle"
+arch="all"
+license="BSD-2-Clause"
+depends="libpuzzle"
```
Comment (andypost) [inline testing/php7-libpuzzle/APKBUILD]: should depend on `php7-common` and probably some other extensions

File: testing/php7-libpuzzle/APKBUILD [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+makedepends="libpuzzle php7-dev"
+source="https://download.pureftpd.org/pub/pure-ftpd/misc/libpuzzle/releases/libpuzzle-$pkgver.tar.gz
+		10_php7_support.patch
+		"
+builddir="$srcdir/libpuzzle-$pkgver"
+options="!check" # Libpuzzle was developed on OpenBSD, and one of the tests doesn't pass because floating point numbers are rounded differently on Linux. It has no implications on usage.
```
Comment (andypost) [inline testing/php7-libpuzzle/APKBUILD]: the oneliner should work for most of php extensions `make NO_INTERACTION=1 REPORT_EXIT_STATUS=1 SKIP_ONLINE_TESTS=1 test`
if there's no tests it should be like `php7 -d extension="$builddir"/modules/libpuzzle.so --ri libpuzzle`
The edge case when extra dependencies needed see `community/php7-pecl-msgpack`

**Comment by andypost** [comment]
@ncopa btw http://www.phash.org/ also looks does not maintain image comparison so only imagemagick & co can do that
